core/src/main/java/com/backblaze/b2/client/B2LargeFileStorer.java

```diff
@@ -0,0 +1,209 @@
+            B2StorageClientWebifier webifier,
+            B2Retryer retryer,
+            Supplier<B2RetryPolicy> retryPolicySupplier,
+            ExecutorService executor) throws B2Exception {
+
+        // Convert the contentSource into a list of B2PartStorer objects.
+        final List<B2PartStorer> partContentSources = new ArrayList<>();
+        try {
+            for (final B2PartSpec partSpec : partSizes.pickParts(contentSource.getContentLength())) {
+                final B2UploadingPartStorer localPartContentSource = new B2UploadingPartStorer(
+                        partSpec.getPartNumber(),
+                        createRangedContentSource(contentSource, partSpec.getStart(), partSpec.getLength()));
+                partContentSources.add(localPartContentSource);
+            }
+        } catch (IOException e) {
+            throw new B2LocalException("trouble", "exception working with content source" + e, e);
+        }
+
+        // Instantiate and return the manager.
+        return new B2LargeFileStorer(
+                largeFileVersion,
+                partContentSources,
+                accountAuthCache,
+                webifier,
+                retryer,
+                retryPolicySupplier,
+                executor);
+    }
+
+    B2FileVersion storeFile() throws B2Exception {
+        final List<Future<B2Part>> partFutures = new ArrayList<>();
+
+        // Store each part in parallel.
+        for (final B2PartStorer partStorer : partStorers) {
+            partFutures.add(executor.submit(() -> partStorer.storePart(this)));
+        }
+
+        return finishLargeFileFromB2PartFutures(fileVersion, partFutures);
+    }
+
+    private B2FileVersion finishLargeFileFromB2PartFutures(B2FileVersion largeFileVersion,
+                                                           List<Future<B2Part>> partFutures) throws B2Exception {
+
+        final List<String> partSha1s = new ArrayList<>();
+        try {
+            for (final Future<B2Part> partFuture : partFutures) {
+                final B2Part part = partFuture.get();
+                partSha1s.add(part.getContentSha1());
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new B2LocalException("interrupted", "interrupted while trying to copy parts: " + e, e);
+
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof B2Exception) {
+                throw (B2Exception) e.getCause();
+            } else {
+                throw new B2LocalException("trouble", "exception while trying to upload parts: " + cause, cause);
+            }
+        } finally {
+            // we've either called get() on all of the futures, or we've hit an exception and
+            // we aren't going to wait for the others.  let's call cancel on all of them.
+            // the ones that have finished already, won't mind and the others will be stopped.
```

Comment:
extra comma in this sentence?  probably copied from my mistaken extra comma in B2LargeFileUploader.  :)

Reply:
Yep, but I took care of it :)